packages/react-storage/src/components/StorageBrowser/actions/handlers/upload.ts

```diff
@@ -46,8 +46,11 @@ export const uploadHandler: UploadHandler = ({
 
   const bucket = constructBucket(config);
 
+  const resolvedKey = isEmpty(data.webkitRelativePath)
+    ? key
+    : data.webkitRelativePath;
```

Comment:
Correct approach but should be applied where the `key` is create and set in the store on file so consumers (override `uploadHandler` functions) don't need to introduce this logic

Reply:
TIL, I think a similar fix could be added to upload cache. Although unrelated to the StoBro work.